internal/build/lifecycle_execution.go

```diff
@@ -720,6 +740,67 @@ func (l *LifecycleExecution) ExtendRun(ctx context.Context, kanikoCache Cache, p
 	return extend.Run(ctx)
 }
 
+const (
+	argBuildID = "build_id"
+	argUserID  = "user_id"
+	argGroupID = "group_id"
+)
+
+/*
+	This implementation of ExtendBuildByDaemon is based on the RFC #0105 which uses docker daemon to extend the build Image instead of kaniko.
+	* Parsing the `group.toml` from the temp directory of buildpack and set the extensions.
+	* Reading the dockerfiles that were generated during the `generate` phase and also parsing the Arguments given by the user from
+	`extend-config.toml`.
+	* Using ImageBuild method of docker API client to extend the Image and save it to the daemon.
+	* Invoking Build phase of lifecycle by creating a container from the extended Image and dropping the privileges.
+
+*/
+
+func (l *LifecycleExecution) ExtendBuildByDaemon(ctx context.Context) error {
+	builderImageName := l.opts.BuilderImage
+	extendedBuilderImageName := l.opts.BuilderImage + "-extended"
```

Comment:
I wonder if it might be better to use a UUID here, to avoid tagging over another image that might already be called `<builder>-extended`. However that might cause the number of images to in the daemon to balloon... I think we can leave it for now unless others feel strongly about it.

Reply:
will add in the help!

Reply:
I'm not totally sure where this would go... maybe a warning when we begin to extend the builder?

Reply:
Is this still outstanding? Some kind of messaging would surely help.

Reply:
done!!
added a warning right after the start of phase

Reply:
> I think we should document this in some place, maybe in the command help?

also added this too! would be surely helpful